sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala

```diff
@@ -47,6 +47,19 @@ trait AnalysisTest extends PlanTest {
     }
   }
 
+  protected def makeAnalyzerWithConf(conf: SQLConf): Analyzer = {
```

Comment:
We cannot test it with `getAnalyzer()` and `withSQLConf` in AnalysisSuite (we need this helper method?)?

Reply:
@maropu The strategy field `fixedPoint` of Analyzer/Optimizer is determined when analyzer/optimizer initialized, thus SQLConf will not affect their behavior during runtime. `getAnalyzer()` only support makeAnalyzer with CASE_SENSITIVE config, while I think adding more config such as ANALYZER_MAX_ITERATIONS to `getAnalyzer()` will be too specific. That’s why I introduced this helper method with SQLConf input parameter, which will also support future test case requirements like ANALYZER_MAX_ITERATIONS does. Thanks!

Reply:
@maropu I moved the helper method in commit ab82c4136144e3bd2b00ce94e87927f7e9eb93f9